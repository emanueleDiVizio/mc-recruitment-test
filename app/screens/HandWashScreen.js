import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import HandWashComponent from '../components/HandWash/HandWashComponent'

const HandWashScreen = () => (
  <View style={styles.container}>
    <Text>Hand Wash Timer</Text>
    <HandWashComponent />
  </View>
)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
})

export default HandWashScreen
