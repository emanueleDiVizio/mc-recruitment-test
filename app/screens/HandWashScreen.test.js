import React from 'react'
import HandWashScreen from './HandWashScreen'
import { render } from '@testing-library/react-native'
import * as redux from 'react-redux'
import MockDate from 'mockdate'

describe('HandWashComponent', () => {
  let state
  let dispatch

  beforeEach(() => {
    dispatch = jest.fn()
    redux.useSelector = selector => selector(state)
    redux.useDispatch = () => dispatch
    MockDate.set('2018-1-1')
  })

  afterEach(() => {
    MockDate.reset()
  })

  it('renders correctly', () => {
    state = {
      shifts: [],
      handWashes: [],
    }
    const { baseElement } = render(<HandWashScreen />)

    expect(baseElement).toMatchSnapshot()
  })
})
