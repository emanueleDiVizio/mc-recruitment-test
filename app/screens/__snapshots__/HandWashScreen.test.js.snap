// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandWashComponent renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "#00A778",
            "flexDirection": "row",
            "height": 100,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 17,
              "marginBottom": 12,
              "textAlign": "center",
            }
          }
        >
          Wash timer
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 21,
              },
              Object {
                "bottom": 12,
                "flexDirection": "row",
                "position": "absolute",
                "right": 12,
              },
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={Object {}}
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "margin": 16,
          }
        }
      >
        <
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00A1D2",
              "borderRadius": 100,
              "flexDirection": "column",
              "height": 200,
              "justifyContent": "center",
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": -2,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 16,
              "width": 200,
            }
          }
          testID="timer-button"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 60,
                },
                Object {
                  "marginBottom": 24,
                },
                Object {
                  "fontFamily": "FontAwesome5Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 21,
              }
            }
          >
            Wash
          </Text>
        </>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#4A4A4A",
                "marginBottom": 8,
                "marginLeft": 16,
              }
            }
          >
            HISTORY
          </Text>
          <ScrollView
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            maxToRenderPerBatch={10}
            onEndReachedThreshold={2}
            removeClippedSubviews={false}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          />
        </View>
      </View>
      <
        style={
          Array [
            Object {
              "backgroundColor": "#00A778",
            },
            Object {
              "bottom": 0,
              "height": 100,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": -4,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 16,
            },
          ]
        }
        testID="shift-button"
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 17,
              "marginTop": 8,
            }
          }
        >
          Start shift
        </Text>
      </>
    </View>
  </View>
</View>
`;
